### DISCUSSION TITLE
What does parallel mean in workflowcore ?

### DISCUSSION THREAD

**Discussion by AngrySKL** [danielgerlag/workflow-core#1239] [Q&A]
I was trying to bulid a workflow which has a parallel task, in the task there are two indepent task which should be run simultaneously.
But after playing around with this [sample](https://github.com/danielgerlag/workflow-core/tree/master/src/samples/WorkflowCore.Sample13) I found the Parallel of WorkflowCore still run tasks in Sequence.

<img width="717" alt="image" src="https://github.com/danielgerlag/workflow-core/assets/8011904/ada5f0a8-88e1-477c-b502-130890004387">

like in above image, the output should not always be the same for each run because the worklfow runs in Parallel, right? But the output is always like below

<img width="236" alt="image" src="https://github.com/danielgerlag/workflow-core/assets/8011904/815d3150-dbf4-4fc8-9b6d-20816baceaab">

the sequence is 1.1 -> 2.1 -> 3.1 -> 1.2 -> 2.2 -> 3.2 -> 2.3 always. The three branchs run in sequence to me.

So either I'm understanding parallel wrong or is there some configuration to make the branchs actually parallel ?

@danielgerlag  could you maybe help to have a look and give me some suggestions? Thanks a lot.

**Comment by AngrySKL** [comment]
Anybody has some thoughts around this?

**Reply by by-2020-github** [reply]
I  also met the same problem , All tasks are executed by the same thread! I can't understand.

**Reply by AngrySKL** [reply]
#785 

It seems to be by design.
